packages/web/src/modules/landing/components/coaches.tsx

```diff
@@ -0,0 +1,105 @@
+import React, { useState, useEffect, ReactNode } from 'react';
+import { ui, components, hooks, requests } from '@application';
+import { css } from '@chess-tent/styled-props';
+import { User } from '@chess-tent/models';
+
+import ViewAllCoachesCard from './components/view-all-coaches-card';
+
+const { Row, Col, Button, Icon, Text } = ui;
+const { CoachCard } = components;
+const { useApi, useIsMobile } = hooks;
+const { publicCoaches } = requests;
+
+const { className: arrowClassName } = css`
+  margin-right: 5px;
+`;
+
+const Coaches = () => {
+  const { fetch: fetchCoaches, response: coachResponse } =
+    useApi(publicCoaches);
+
+  const isMobile = useIsMobile();
+  const cardStep = isMobile ? 1 : 3;
+
+  const [cardIndex, setCardIndex] = useState(cardStep);
+  const [cards, setCards] = useState<ReactNode[]>([]);
+  const [coaches, setCoaches] = useState<User[]>([]);
+
+  const onLeftArrowClick = () => {
+    setCardIndex(prevIndex =>
+      prevIndex > cardStep ? prevIndex - cardStep : cards.length,
+    );
+  };
+
+  const onRightArrowClick = () => {
+    console.log({ cardIndex });
+    setCardIndex(prevIndex =>
+      prevIndex <= coaches.length ? prevIndex + cardStep : cardStep,
+    );
+  };
+
+  useEffect(() => {
+    fetchCoaches();
+  }, [fetchCoaches]);
+
+  useEffect(() => {
+    if (coaches.length > 0 && coachResponse?.data) {
```

Comment:
This isn't good.

You don't want to set elements into state. Based on the current index you can slice coaches ad-hoc and map them to card elements inside the render.

Reply:
Had it like that then I change to this for some reason, will return